libtransmission/verify.cc

```diff
@@ -83,23 +102,59 @@ void tr_verify_worker::verify_torrent(Mediator& verify_mediator, std::atomic<boo
         piece_pos += bytes_this_pass;
         file_pos += bytes_this_pass;
 
+        if (left_in_file == 0)
+        {
+            /* We want all on_file_error() calls for a file index to be consecutive. Therefore
+             * don't add files for which an error has already been emitted. We don't need to report
+             * piece errors for missing or wrong length files.
+             */
+            if (file_valid)
+            {
+                files_ending_in_current_piece.push_back(file_index);
```

Comment:
if file X _begins_ in the current piece but doesn't end in the piece, and the piece checksum fails, will `on_file_error()` be called for file X?

Reply:
As per my previous discussion with the author, it will here:

https://github.com/transmission/transmission/blob/0f321b850342646c8ca3fd5e54a75220e34865cf/libtransmission/verify.cc#L136-L139